src/lib.rs

```diff
@@ -523,12 +531,14 @@ pub enum AuthType {
 /// Stores the configuration for an OAuth2 client.
 ///
 #[derive(Clone, Debug)]
-pub struct Client<TE, TR, TT, TIR>
+pub struct Client<RT, TE, TR, TT, TIR, RER>
 where
+    RT: RevocableToken,
     TE: ErrorResponse,
     TR: TokenResponse<TT>,
     TT: TokenType,
     TIR: TokenIntrospectionResponse<TT>,
+    RER: ErrorResponse,
```

Comment:
now that we have two `ErrorResponse` type parameters, it's probably time to document them in the doc comment. these were probably self-evident before, but less so now

Reply:
I wondered if it would be better to not even put the additional types at the struct level but only on the revoke fn, that way you don't break existing code that already has custom `Client`s... ?

Reply:
I think there's been some related discussion in the past. if we put the type arguments on the function instead, that makes the call sites more complex since users will need to add type annotations there. the tradeoff here is that we have a more complex `Client` type definition with backward compatibility issues, but once you define a type alias for `Client`, the call sites become simpler

Reply:
Understood.

Reply:
>now that we have two ErrorResponse type parameters, it's probably time to document them in the doc comment. these were probably self-evident before, but less so now

I pushed something for this, but I'm not convinced this is really what you had in mind.

Reply:
thanks for expanding the docs! I think they address my concerns around `TE`/`TRE`